x/permission/types/genesis.go

```diff
@@ -0,0 +1,54 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// NewGenesisState is the constructor for GenesisState
+func NewGenesisState(governanceKey *multisig.LegacyAminoPubKey, accounts []GovAccount) *GenesisState {
+	return &GenesisState{
+		GovernanceKey: governanceKey,
+		GovAccounts:   accounts,
+	}
+}
+
+// DefaultGenesisState returns a genesis state with default parameters
+func DefaultGenesisState() *GenesisState {
+	return NewGenesisState(nil, []GovAccount{})
+}
+
+// Validate performs a validation check on the genesis parameters
+func (m GenesisState) Validate() error {
+	if m.GovernanceKey != nil {
+		if int(m.GovernanceKey.Threshold) > len(m.GovernanceKey.PubKeys) {
+			return fmt.Errorf("threshold k of n multisignature: len(pubKeys) < k")
+		}
+	}
+
+	for _, account := range m.GovAccounts {
+		if err := account.Validate(); err != nil {
+			return getValidateError(err)
+		}
+	}
```

Comment:
also check there is exactly one account with role governance and that it matches the key

Reply:
I had trouble to unpack the proto any here, can only check exact one ROLE_ACCESS_CONTROL account for now